### DISCUSSION TITLE
Why switch from subpath exports to barrel files?

### DISCUSSION THREAD

**Discussion by justin0mcateer** [libp2p/js-libp2p#2310] [Q&A] [answered]
When the mono-repo was created, the exports under the libp2p package were specified using 'exports' in package.json. With 1.0, these subpath exports have been eliminated in favor of barrel file. Why so?

**Comment by achingbrain** [comment, marked as answer]
It was getting increasingly hard to remember which interfaces were exported from which sub paths, and some of them were inconsistent and arbitrary, e.g:

```TypeScript
import type { Connection } from '@libp2p/interface/connection'

// vs

import type { Stream } from '@libp2p/interface/stream-muxer'
```

so for V1 everything has shifted to the root which is a bit simpler to consume:

```TypeScript
import type { Connection, Stream } from '@libp2p/interface'
```
